### ISSUE TITLE
Paf.cpp, index out of range error

### ISSUE THREAD

**Issue by aaron-michaux** [tensorlayer/hyperpose#318]
There seems to be an error with our indices are handled when parsing features. In particular, in the function `get_humans`

```
static std::vector<human_ref_t>
   get_humans(const std::vector<peak_info>& all_peaks,
                        const std::vector<std::vector<connection>>& all_connections)
{
   ...
         auto& hr1 = human_refs[hr_ids[0]];
         auto& hr2 = human_refs[hr_ids[1]]; // See note below
   ...
}
```

If you change the marked line above to:

```
auto& hr2 = human_refs.at(hr_ids[1]); // Index out of range!
```

So somewhere the `hr_ids[...]` are getting borked.

**Comment by korejan** [comment]
This is a legitimate bug that's existed since version 1, yes when an element is deleted that's not the last element all the preexisting index references past that element need updating. I've fixed in my local repository and I kept forgetting to send a pull request.

**Comment by ganler** [comment]
Hi @korejan , could send us a PR about that if convinient?

**Comment by ganler** [comment]
@korejan Hi, are you available to send us a PR about this? Or you can indicate where you made such modification and I can adopt it manually.

**Comment by korejan** [comment]
> 
> 
> @korejan Hi, are you available to send us a PR about this? Or you can indicate where you made such modification and I can adopt it manually.

@ganler sorry I've been extremly busy at work in the past couple of months. The fix is very simple though the reason why I haven't submitted a PR is that there's actually two ways to fix it, one will be slightly faster for (very) large vectors but it makes the assumption that the human_refs list elements are stored in order, if i remember correctly when i was debugging this a couple of months ago i think this was the case but I don't know if this was intended or not so my local fix iterates through all the entries, my change does this in paf.cpp/get_humans:

![2020-10-08 19_29_51-GitHub Desktop](https://user-images.githubusercontent.com/64199710/95499693-95a3fd80-099d-11eb-8425-d97c0d600cf5.png)

**Comment by ganler** [comment]
@aaron-michaux Hi, could you provide a sample that can reproduce this error? We are working on a fix on this and I think I need to test the fix.

**Comment by aaron-michaux** [comment]
Hi, I'll try to get something. Sorry, just noticed this. I've also found some other bugs, which I didn't bother to investigate. (Maybe I can send you problem images?)

**Comment by ganler** [comment]
> Hi, I'll try to get something. Sorry, just noticed this. I've also found some other bugs, which I didn't bother to investigate. (Maybe I can send you problem images?)

Yes, please. Just report the errors and bugs and we will try our best to fix them..

**Comment by aaron-michaux** [comment]
How do I send you an image. 

It's propriety -- no big deal if this particular image finds its way onto the interwebs, but if I send you _other_ images, then they'll need to be kept private.

Please advise.

**Comment by ganler** [comment]
@aaron-michaux Through my e-mail: jaway.liu@gmail.com

**Comment by ganler** [comment]
I tried many models but did not found any segmentation fault. Could you also indicate which model you are using?
